### DISCUSSION TITLE
Can the .NET 7 -> .NET 8 migration plan be made clearer?

### DISCUSSION THREAD

**Discussion by RobTF** [dotnet/maui#16847] [General] [answered]
Hi,

I stumbled across this yesterday on the .NET MAUI blog; 

![image](https://github.com/dotnet/maui/assets/4292237/be2222fa-c2e1-48df-9027-b25c8a4b84a9)

The last phrase is what really caught my attention - "We encourage everyone to consider using .NET 8 releases from this point forward." with the note of a " go live support license" from .NET 8 RC1.

This is major, major news, yet is posted as a kind of side-note on a blog post. Is there somewhere this can be put where it is more permanent/prominent? Should developers new to MAUI now be ignoring .NET 7 and heading straight for .NET 8?

For example, the main news links on the Github project homepage (which many more people will inevitably hit) still shows old articles and hasn't been updated in some time.

**Comment by MartyIX** [comment]
> The last phrase is what really caught my attention - "We encourage everyone to consider using .NET 8 releases from this point forward." with the note of a " go live support license" from .NET 8 RC1.

I believe that "go live support license" is sort of common for .NET products' RC versions. So it's not specifically a .NET MAUI thing.

> Should developers new to MAUI now be ignoring .NET 7 and heading straight for .NET 8?

I'm watching .NET MAUI development and I tend to think that .NET 8 version is much better and one would spend much less time in looking for workarounds for issues. So for new projects or people new to .NET MAUI, I think it makes good sense, yes.

**Reply by RobTF** [reply]
I see, do you know if it is possible to run .NET 7 and 8 workloads side by side? We have a relatively stable build of our app on .NET 7 but I'd like to see if any of the bug fixes in .NET 8 will help with some of our issues without blowing up my dev environment by messing up the .NET 7 install.

**Reply by MartyIX** [reply]
Yes, you just need to install .NET 8 MAUI using these instructions https://github.com/dotnet/maui/releases/tag/8.0.0-preview.7.8842. I'm testing on .NET 8 often to see if a bug is fixed or not.

It might be good to have something like this in your csproj file:

```xml
<PropertyGroup>
  <!-- Either "preview" (.NET 8) or "stable" (.NET 7) (invalid values are mapped to this value) -->
  <MauiVersionSelector>stable</MauiVersionSelector>
</PropertyGroup>

<PropertyGroup Condition="$(MauiVersionSelector) == 'preview'">
  <MauiVersion>8.0.0-preview.7.8842</MauiVersion>
</PropertyGroup>

<ItemGroup Condition="$(MauiVersionSelector) == 'preview'">
  <PackageVersion Include="Microsoft.Maui.Controls" Version="$(MauiVersion)" />
  <PackageVersion Include="Microsoft.Maui.Controls.Compatibility" Version="$(MauiVersion)" />
</ItemGroup>
```

**Reply by RobTF** [reply]
Great, thanks that's really good to know.

One last thing... Internally we have teams which distribute NuGet packages to an internal source and these target .NET 7. Our apps make heavy use of these packages. Can a .NET 8 project reference these without change or would I be blocked waiting for those teams to provide a .NET 8 version of any packages I need first? I know I could try it but I'm trying to assess whether it's worth the time/risk to start messing with it yet.

**Reply by MartyIX** [reply, marked as answer]
You can use .NET 7 nuget packages without an issue. Or at least I have not experienced any issue.

**Reply by RobTF** [reply]
Great, thanks for the heads up on this.

**Comment by maonaoda** [comment]
> 是的，您只需使用这些说明安装 .NET 8 MAUI https://github.com/dotnet/maui/releases/tag/8.0.0-preview.7.8842。我经常在 .NET 8 上进行测试，以查看错误是否已修复。
> 
> 在你的csproj文件中有这样的东西可能会很好：
> 
> ```
> <PropertyGroup>
>   
>   <MauiVersionSelector>stable</MauiVersionSelector>
> </PropertyGroup>
> 
> <PropertyGroup Condition="$(MauiVersionSelector) == 'preview'">
>   <MauiVersion>8.0.0-preview.7.8842</MauiVersion>
> </PropertyGroup>
> 
> <ItemGroup Condition="$(MauiVersionSelector) == 'preview'">
>   <PackageVersion Include="Microsoft.Maui.Controls" Version="$(MauiVersion)" />
>   <PackageVersion Include="Microsoft.Maui.Controls.Compatibility" Version="$(MauiVersion)" />
> </ItemGroup>
> ```

very useful,thanks so much
